js/Integer.js

```diff
@@ -0,0 +1,64 @@
+const InvalidInput = require("./Errors")
+
+class Integer {
+    static plus(a, b) {
+        Integer.#validate(a, b)
+
+        let carry = 0
+        while (b !== 0) {
+            carry = a & b
+            a = a ^ b
+            b = carry << 1
+        }
+
+        return a
+    }
+
+    static product(a, b) {
+        Integer.#validate(a, b)
+
+        let signedBit = 1 << 31
```

Comment:
1. This is not a variable but a constant. Use the `const` keyword instead.
2. I don't like the name. This is not a individual bit, but a complete Number object. Something like `minInteger` would be better (or `MIN_INTEGER` if you prefer all caps for constants, I'll leave that for you to decide.).